src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/EligibilityService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.companieshouse.confirmationstatementapi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.confirmationstatementapi.ConfirmationStatementApiApplication;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityRule;
+import uk.gov.companieshouse.confirmationstatementapi.exception.EligibilityException;
+import uk.gov.companieshouse.confirmationstatementapi.model.response.CompanyValidationResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class EligibilityService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfirmationStatementApiApplication.APP_NAME);
+    private final List<EligibilityRule<CompanyProfileApi>> eligibilityRules;
+
+    @Autowired
+    public EligibilityService(@Qualifier("confirmation-statement-eligibility-rules")
+                                                    List<EligibilityRule<CompanyProfileApi>> eligibilityRules){
+        this.eligibilityRules = eligibilityRules;
+    }
+
+    public CompanyValidationResponse checkCompanyEligibility(CompanyProfileApi companyProfile) {
+        CompanyValidationResponse response = new CompanyValidationResponse();
+        try {
+            for (EligibilityRule<CompanyProfileApi> eligibilityRule : eligibilityRules) {
+                eligibilityRule.validate(companyProfile);
+            }
+        } catch (EligibilityException e) {
+            LOGGER.info(String.format("Company %s ineligible to use the service because %s", companyProfile.getCompanyNumber(), e.getEligibilityStatusCode().toString()));
+            response.setValidationError(e.getEligibilityStatusCode());
+            return response;
+        }
+        return response;
```

Comment:
if it passes all validation the response.eligibilityStatusCode should be set to CompanyValidForService